src/hooks/useConvert.jsx

```diff
@@ -1,72 +1,78 @@
 import React from 'react';
+import { parseCSV, stringifyCSV, parseJson, generateFileName, fetchGeoData, insertGeoData } from '../utils/csv';
 
 const useConvert = () => {
   const [file, setFile] = React.useState(undefined);
-  const [isConvertToJson, setIsConvertToJson] = React.useState(false);
-  const [jsonData, setJsonData] = React.useState([]);
+  const [csvText, setCsvText] = React.useState('');
+  const [jsonText, setJsonText] = React.useState('');
 
   const fileReader = new FileReader();
 
   const handleUploadFileChange = (event) => {
     setFile(event.target.files[0]);
-    setIsConvertToJson(false);
-  };
-
-  const convertCsvFileToJson = React.useCallback((text) => {
-    const csvHeader = text.slice(0, text.indexOf('\n')).replace('\r', '').split(',');
-    let csvData = text
-      .slice(text.indexOf('\n') + 1)
-      .split('\n')
-      .map((s) => s.replace('\r', '').split(','));
+    fileReader.onload = function (event) {
+      const text = event.target.result;
+      setCsvText(text);
+    };
 
-    csvData.splice(-1);
-
-    const convertData = csvData.map((data) => {
-      let newObject = {};
-      csvHeader.forEach((header, index) => {
-        newObject = { ...newObject, [header]: data[index] };
-      });
-      return newObject;
-    });
+    fileReader.readAsText(event.target.files[0]);
+  };
 
-    setJsonData(convertData);
+  const convertCsvFileToJson = React.useCallback((csvText) => {
+    const { csvHeader, csvData } = parseCSV(csvText);
+    const json = stringifyCSV({ csvHeader, csvData });
+    setJsonText(json);
```

Comment:
You don't need the csv header for your use case, I would change that to:
```suggestion
    const data = parseCSV(csvText);
    const jsonText = JSON.stringify(data);
    setJsonText(jsonText);
```
This has the advantage that `parseCSV` has the same signature as [`JSON.parse`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/parse#return_value) and you don't need your `stringifyCSV`. Optionally you could add another function that is returning the CSV header, but that is not needed for this use case